### DISCUSSION TITLE
No connection to Bluetooth, PSRAM ID read error

### DISCUSSION THREAD

**Discussion by eypeop** [pschatzmann/ESP32-A2DP#625] [Q&A] [answered]
Hello, I'm new with microcontroller programming, but I want to do a Spotify Speaker, so I found your library and try it.
I have ESP32-WROOM Dev Board, work in Arduino IDE, so try to upload a code with internal DAC (example: bt_music_receiver_to_internal_dac.ino), it's uploaded, but in serial monitor was an error:

19:51:14.776 -> rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
19:51:14.776 -> configsip: 0, SPIWP:0xee
19:51:14.820 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
19:51:14.820 -> mode:DIO, clock div:1
19:51:14.820 -> load:0x3fff0030,len:4832
19:51:14.820 -> load:0x40078000,len:16460
19:51:14.820 -> load:0x40080400,len:4
19:51:14.820 -> load:0x40080404,len:3504
19:51:14.820 -> entry 0x400805cc
19:51:15.394 -> E (621) quad_psram: PSRAM ID read error: 0xffffffff, PSRAM chip not found or not supported

And no connection with Bluetooth (Can't see the ESP on my phone/laptop), but when I try classic example of Bluetooth Serial library - > I can connect with ESP32 by my laptop.
And I search in the Internet, a lot of people just select another board, like ESP32-WROOM to ESP32 Dev Module, where in Tools I can see variant of "PSRAM" -> Enabled or Disabled (both variants don't work)
My configuration:
![image_2024-11-06_195543736](https://github.com/user-attachments/assets/5f1223d9-0456-4307-8f69-db50239cc262)

I don't know what else I can do... I will be glad for any help ;(

**Comment by pschatzmann** [comment, marked as answer]
WROVERS have psram, WROOMS do not.
So do not activate psram in the Tools menu!

**Reply by eypeop** [reply]
> WROVERS have psram, WROOMS do not. So do not activate psram in the Tools menu!

it doesn't work at all. the same error.  Library cannot work with ESP32 WROOM?

**Reply by pschatzmann** [reply]
Yes it does: but try some simple hello world sketch first!

**Reply by eypeop** [reply]
Everything is good...

![image](https://github.com/user-attachments/assets/07ecc54f-a613-4eee-b258-a7f3caa6b592)

**Reply by pschatzmann** [reply]
Don't forget to set the Core Debug Level to activate the logging...

**Reply by eypeop** [reply]
Okey, I can see thar no PSRAM on the board, but what else? Why does simple code example from lib not work?
Chip Info:
20:36:55.331 -> ------------------------------------------
20:36:55.331 ->   Model             : ESP32
20:36:55.365 ->   Package           : D0WD-Q5
20:36:55.365 ->   Revision          : 3.01
20:36:55.365 ->   Cores             : 2
20:36:55.365 ->   CPU Frequency     : 240 MHz
20:36:55.365 ->   XTAL Frequency    : 40 MHz
20:36:55.365 ->   Features Bitfield : 0x00000032
20:36:55.397 ->   Embedded Flash    : No
20:36:55.397 ->   Embedded PSRAM    : No
20:36:55.397 ->   2.4GHz WiFi       : Yes
20:36:55.397 ->   Classic BT        : Yes
20:36:55.397 ->   BT Low Energy     : Yes
20:36:55.397 ->   IEEE 802.15.4     : No
20:36:55.397 -> ------------------------------------------
20:36:55.432 -> INTERNAL Memory Info:
20:36:55.432 -> ------------------------------------------
20:36:55.432 ->   Total Size        :   380612 B ( 371.7 KB)
20:36:55.432 ->   Free Bytes        :   349580 B ( 341.4 KB)
20:36:55.432 ->   Allocated Bytes   :    23788 B (  23.2 KB)
20:36:55.432 ->   Minimum Free Bytes:   344148 B ( 336.1 KB)
20:36:55.466 ->   Largest Free Block:   118772 B ( 116.0 KB)
20:36:55.466 -> ------------------------------------------
20:36:55.466 -> Flash Info:
20:36:55.466 -> ------------------------------------------
20:36:55.466 ->   Chip Size         :  4194304 B (4 MB)
20:36:55.466 ->   Block Size        :    65536 B (  64.0 KB)
20:36:55.499 ->   Sector Size       :     4096 B (   4.0 KB)
20:36:55.499 ->   Page Size         :      256 B (   0.2 KB)
20:36:55.499 ->   Bus Speed         : 80 MHz
20:36:55.499 ->   Bus Mode          : QIO

**Comment by pschatzmann** [comment]
Is this the log with the A2DP sketch ?

**Reply by eypeop** [reply]
![image](https://github.com/user-attachments/assets/4fff15ab-ed4b-4378-81eb-f9d4fda5489d)
![image](https://github.com/user-attachments/assets/238833c8-270b-4a06-9f1e-d8b2cf7d764a)
![image](https://github.com/user-attachments/assets/9966c2e5-c518-41f0-9c92-253fef8baffc)

**Reply by pschatzmann** [reply]
So, it is working: I don't see any errors

**Reply by eypeop** [reply]
Yes, but no connections with Bluetooth of my phone or laptop

**Reply by pschatzmann** [reply]
Please copy and paste the complete log and don't use any screen shots: this is impossible to read

**Reply by pschatzmann** [reply]
Also explain what sketch you used!

**Reply by pschatzmann** [reply]
I was using the bt_music_receiver_to_internal_dac

This was the expected log (with log level info)
```
load:0x40078000,len:16460
load:0x40080400,len:4
load:0x40080404,len:3504
entry 0x400805cc
[    36][I][BluetoothA2DPCommon.cpp:257] log_free_heap(): [BT_AV] Available Heap: 196752
[    44][I][BluetoothA2DPSink.cpp:101] start(): [BT_AV] Device name will be set to 'InternalDAC'
[    55][I][BluetoothA2DPCommon.cpp:177] get_last_connection(): [BT_AV] nvs_blob does not exist
[    65][I][BluetoothA2DPSink.cpp:164] init_i2s(): [BT_AV] init_i2s
[    73][I][BluetoothA2DPOutput.cpp:196] begin(): [BT_AV] begin
[   147][I][BluetoothA2DPSink.cpp:189] init_bluetooth(): [BT_AV] controller initialized
[   468][I][BluetoothA2DPSink.cpp:198] init_bluetooth(): [BT_AV] bluedroid initialized
[   541][I][BluetoothA2DPSink.cpp:207] init_bluetooth(): [BT_AV] bluedroid enabled
[   548][I][BluetoothA2DPCommon.cpp:358] set_scan_mode_connectable(): [BT_AV] set_scan_mode_connectable true
[   556][I][BluetoothA2DPSink.cpp:431] app_gap_callback(): [BT_AV] event: 10
[   566][I][BluetoothA2DPSink.cpp:514] av_hdl_a2d_evt(): [BT_AV] A2DP PROF STATE: Init Compl

[   566][I][BluetoothA2DPSink.cpp:431] app_gap_callback(): [BT_AV] event: 10
[   584][I][BluetoothA2DPSink.cpp:431] app_gap_callback(): [BT_AV] event: 10
[   584][I][BluetoothA2DPSink.cpp:158] start(): [BT_AV] IDF Version 5.1
[   599][I][BluetoothA2DPSink.cpp:431] app_gap_callback(): [BT_AV] event: 10
[   599][I][BluetoothA2DPCommon.cpp:257] log_free_heap(): [BT_AV] Available Heap: 56200
```
And the Available Devices was showing  "Internal DAC".
Maybe you just need to scroll down to see it ?

**Reply by eypeop** [reply]
21:51:20.946 -> mode:DIO, clock div:1
21:51:20.946 -> load:0x3fff0030,len:4832
21:51:20.946 -> load:0x40078000,len:16460
21:51:20.946 -> load:0x40080400,len:4
21:51:20.946 -> load:0x40080404,len:3504
21:51:20.946 -> entry 0x400805cc
21:51:21.546 -> [     1][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RX (2) successfully set to 0x400e3ba8
21:51:21.577 -> [    12][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_TX (3) successfully set to 0x400e3b78
21:51:21.577 -> [    26][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_CTS (4) successfully set to 0x400e3b48
21:51:21.616 -> [    39][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RTS (5) successfully set to 0x400e3b18
21:51:21.616 -> [    53][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RX (2) successfully set to 0x400e3ba8
21:51:21.616 -> [    66][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_TX (3) successfully set to 0x400e3b78
21:51:21.645 -> [    80][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_CTS (4) successfully set to 0x400e3b48
21:51:21.645 -> [    93][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RTS (5) successfully set to 0x400e3b18
21:51:21.682 -> [   107][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RX (2) successfully set to 0x400e3ba8
21:51:21.682 -> [   120][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_TX (3) successfully set to 0x400e3b78
21:51:21.682 -> [   133][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_CTS (4) successfully set to 0x400e3b48
21:51:21.714 -> [   147][V][esp32-hal-periman.c:235] perimanSetBusDeinit(): Deinit function for type UART_RTS (5) successfully set to 0x400e3b18
21:51:21.714 -> E (794) quad_psram: PSRAM ID read error: 0xffffffff, PSRAM chip not found or not supported
21:51:21.714 -> [   162][W][esp32-hal-psram.c:67] psramInit(): [arduino-psram] PSRAM init failed!
21:51:21.745 -> [   192][D][esp32-hal-cpu.c:264] setCpuFrequencyMhz(): PLL: 480 / 2 = 240 Mhz, APB: 80000000 Hz
21:51:21.779 -> [   201][E][esp32-hal-psram.c:90] psramAddToHeap(): PSRAM not initialized!
21:51:21.779 -> [   226][V][esp32-hal-periman.c:160] perimanSetPinBus(): Pin 3 successfully set to type UART_RX (2) with bus 0x3ffbdb80
21:51:21.813 -> [   236][V][esp32-hal-periman.c:160] perimanSetPinBus(): Pin 1 successfully set to type UART_TX (3) with bus 0x3ffbdb80
21:51:22.276 -> [   731][D][BluetoothA2DPSink.cpp:90] start(): [BT_AV] start
21:51:22.308 -> [   736][I][BluetoothA2DPCommon.cpp:257] log_free_heap(): [BT_AV] Available Heap: 195768
**21:51:22.308 -> [   746][I][BluetoothA2DPSink.cpp:102] start(): [BT_AV] Device name will be set to 'InternalDAC'**
21:51:22.308 -> [   757][D][BluetoothA2DPSink.cpp:1063] init_nvs(): [BT_AV] init_nvs
21:51:22.344 -> [   765][D][BluetoothA2DPCommon.cpp:163] get_last_connection(): [BT_AV] get_last_connection
21:51:22.344 -> [   776][I][BluetoothA2DPCommon.cpp:177] get_last_connection(): [BT_AV] nvs_blob does not exist
21:51:22.344 -> [   786][D][BluetoothA2DPCommon.cpp:186] get_last_connection(): [BT_AV] => 00:00:00:00:00:00
21:51:22.390 -> [   797][I][BluetoothA2DPSink.cpp:165] init_i2s(): [BT_AV] init_i2s
21:51:22.390 -> [   805][I][BluetoothA2DPOutput.cpp:199] begin(): [BT_AV] begin
21:51:22.390 -> [   813][D][BluetoothA2DPSink.cpp:186] init_bluetooth(): [BT_AV] init_bluetooth
21:51:22.484 -> [   896][I][BluetoothA2DPSink.cpp:191] init_bluetooth(): [BT_AV] controller initialized
21:51:22.796 -> [  1213][I][BluetoothA2DPSink.cpp:200] init_bluetooth(): [BT_AV] bluedroid initialized
21:51:22.830 -> [  1285][I][BluetoothA2DPSink.cpp:209] init_bluetooth(): [BT_AV] bluedroid enabled
21:51:22.864 -> [  1293][D][BluetoothA2DPSink.cpp:322] app_task_start_up(): [BT_AV] app_task_start_up
21:51:22.864 -> [  1303][D][BluetoothA2DPSink.cpp:1178] ccall_app_task_handler(): [BT_AV] ccall_app_task_handler
21:51:22.864 -> [  1313][D][BluetoothA2DPSink.cpp:278] app_task_handler(): [BT_AV] app_task_handler
21:51:22.897 -> [  1323][D][BluetoothA2DPSink.cpp:234] app_work_dispatch(): [BT_API] app_work_dispatch event 0x0, param len 0
21:51:22.897 -> [  1335][D][BluetoothA2DPSink.cpp:264] app_send_msg(): [BT_AV] app_send_msg
21:51:22.897 -> [  1344][D][BluetoothA2DPSink.cpp:286] app_task_handler(): [BT_API] app_task_handler, sig 0x1, 0x0
21:51:22.930 -> [  1355][D][BluetoothA2DPSink.cpp:289] app_task_handler(): [BT_API] app_task_handler, APP_SIG_WORK_DISPATCH sig: 1
21:51:22.930 -> [  1367][D][BluetoothA2DPSink.cpp:257] app_work_dispatched(): [BT_AV] app_work_dispatched
21:51:22.930 -> [  1377][D][BluetoothA2DPSink.cpp:1211] ccall_av_hdl_stack_evt(): [BT_AV] ccall_av_hdl_stack_evt
21:51:22.965 -> [  1388][D][BluetoothA2DPSink.cpp:911] av_hdl_stack_evt(): [BT_AV] av_hdl_stack_evt evt 0
21:51:22.965 -> [  1398][D][BluetoothA2DPSink.cpp:916] av_hdl_stack_evt(): [BT_AV] av_hdl_stack_evt av_hdl_stack_evt BT_APP_EVT_STACK_UP
21:51:22.965 -> [  1411][D][BluetoothA2DPSink.cpp:926] av_hdl_stack_evt(): [BT_AV] AVRCP controller initialized!
21:51:22.997 -> [  1411][D][BluetoothA2DPSink.cpp:1205] ccall_app_gap_callback(): [BT_AV] ccall_app_gap_callback
21:51:22.997 -> [  1430][I][BluetoothA2DPSink.cpp:433] app_gap_callback(): [BT_AV] event: 10
21:51:22.997 -> [  1430][D][BluetoothA2DPSink.cpp:972] av_hdl_stack_evt(): [BT_AV] set_scan_mode_connectable(true)
21:51:22.997 -> [  1448][I][BluetoothA2DPCommon.cpp:358] set_scan_mode_connectable(): [BT_AV] set_scan_mode_connectable true
21:51:23.031 -> [  1452][D][BluetoothA2DPSink.cpp:1191] ccall_app_a2d_callback(): [BT_AV] ccall_app_a2d_callback
21:51:23.031 -> [  1468][D][BluetoothA2DPSink.cpp:986] app_a2d_callback(): [BT_AV] app_a2d_callback
21:51:23.031 -> [  1468][I][BluetoothA2DPSink.cpp:159] start(): [BT_AV] IDF Version 5.1

it must work, i see, but it is what it is
